### ISSUE TITLE
Add support for implicit member expression

### ISSUE THREAD

**Issue by TheTripleV** [microsoft/pylance-release#4735]
Swift has support for [implicit member expression](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/expressions/#Implicit-Member-Expression) (also called leading dot notation).

In python terms, the following codes would be valid
```python
class Color(Enum):
	RED = 1

mycolor: Color = .RED
```
Static members can be looked up by type inference. Since mycolor expects a `Color`, `.RED` resolves to `Color.RED`.

```python
class Color(Enum):
	RED = 1

	@staticmethod
	def random_color() -> "Color": ...

mycolor2: Color = .random_color()
```
Static methods can also be looked up.

```python
def print_color(color: Color): ...

print_color(.RED)
print_color(.random_color())
```
This works for function arguments too.

This feature cannot be added to Python itself because if relies on type information.
But, it would be nice if it could be added to Pylance.

If pyright knows the types allowed, and the user presses the dot key in an empty field,
1. Pylance can suggest valid members for all valid types
2. If the user presses enter or continues lookup by pressing dot, Pylance fills in the complete type

**Comment by erictraut** [comment]
This is not valid Python syntax. If you attempt to run any of the above code samples in the Python interpreter, you'll see that they all result in a syntax error. 

If you want to propose changes to the Python language, the [python "ideas"](https://discuss.python.org/c/ideas/6 discussion) would be a more appropriate forum.

**Comment by TheTripleV** [comment]
Yes, that code isn't valid. I'm asking for the dot key to enable more completions within Pylance.

### Current
Currently, given
```python
from enum import Enum

class Color(Enum):
	RED = 1

mycolor: Color = Color.
```
the moment I press the `.`, I get a suggestion for `RED`
<img width="244" alt="Screenshot 2023-08-21 at 4 10 45 PM" src="https://github.com/microsoft/pylance-release/assets/11065157/50919fbb-402c-4d63-bd24-cacfdd116da3">



Starting at `mycolor: Color = `, the keystrokes I use to type a valid value are `['Shift+c', 'Enter', '.', 'Enter']`.

### Request
It would be nice if in addition to that, given
```python
from enum import Enum

class Color(Enum):
	RED = 1

mycolor: Color = 
```

I can press `.`, and I'll be suggested `Color.RED` in its entirety. 
The keystrokes I would use would be `['.', 'Enter']`.
Because pyright knows the type is `Color`, it can fill in that information for me.

I hope this explains my request better. I understand that in this basic example, its two keystrokes saved. But, the end result is that I no longer have to thing about how a developer organizes / lays out the types in their library. The type in question could be multiple modules / nested classes deep, but Pylance would fill it in for me.

If a library had a `controllerlib.HIDS.Controller.Hand` enum with types `Left` and `Right`, and had a toplevel `controllerlib.vibrate(hand: "the above type")`, then typing a single `.` and having the correct type filled in would be way better than going hunting for the full type path required.

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
